configure.ac

```diff
@@ -328,6 +328,11 @@ SQUID_CC_ADD_CXXFLAG_WARNING_IF_SUPPORTED([-Wshadow])
 SQUID_CC_ADD_CXXFLAG_WARNING_IF_SUPPORTED([-Wmissing-declarations])
 SQUID_CC_ADD_CXXFLAG_WARNING_IF_SUPPORTED([-Woverloaded-virtual])
 
+dnl MacOS provides LDAP and SASL that use deprecated declarations
+AS_IF([test "$squid_host_os" = "darwin"],[
+  SQUID_CC_ADD_CXXFLAG_WARNING_IF_SUPPORTED([-Wno-deprecated-declarations])
```

Comment:
After this PR and other MacOS portability fixes, will Squid _always_ enable LDAP and SASL support on MacOS? If not, is there a way to make this `-Wno-deprecated-declarations` addition _conditional_ as well, so that we do not have to silence useful warnings in other headers on MacOS.

Reply:
No, the normal behaviours are preserved, e.g. if an administrator decides to not build LDAP helpers, they will not be built.
There are two locations where these headers may be found on MacOS:
- as part of Xcode (where the API we use are marked as deprecated)
- as part of homebrew (where it's the standard openldap  and cyrus-SASL)

With this flag, if the Xcode headers are installed, LDAP squid helpers will build; without it, they won't.

I don't understand the logic on making using this flag conditional: Squid's default behaviour across all systems is to build all helpers that can be built, this flag restores that default behaviour.

At this time, these are the only deprecated APIs we are relying on (checked by grep-ing the output of a build on MacOS for the keyword 'deprecated')

Reply:
> I don't understand the logic on making using this flag conditional: 

We do not want to disable useful warnings in environments where disabling them does not address any known problems.

> Squid's default behaviour across all systems is to build all helpers that can be built

I do not want to change that default. I also want to preserve/enable LDAP-from-Xcode support where possible. This change request is not about any of that.

If possible, please disable these warnings (i.e. add `-Wno-deprecated-declarations`) only when LDAP helpers are enabled on MacOS.

Reply:
done